### PR TITLE
[ACTION] Helper Function: Random Item: return multiple

### PR DESCRIPTION
**Pull request by zalmanlew**
## WHAT
Helper Function **Random Item from List** action: allow returning multiple random items from the list.

## WHY
Sometimes the user will want to return multiple random items from the given list.

## HOW

The default (and minimum) value will remain at 1 for the out-of-the-box solution to work as it does now, if the user wants they can specify a higher quantity.

### <samp>🤖 Generated by Copilot at ef8ed71</samp>

* Update the action name, description, and version to support returning multiple random items from the list ([link](https://github.com/PipedreamHQ/pipedream/pull/6233/files?diff=unified&w=0#diff-ddc1f2837cd6e660224736fb43797210ff92faa88f5290c3fd483952e8cd076bL2-R9))
* Add a quantity prop to let the user specify how many items to return ([link](https://github.com/PipedreamHQ/pipedream/pull/6233/files?diff=unified&w=0#diff-ddc1f2837cd6e660224736fb43797210ff92faa88f5290c3fd483952e8cd076bL16-R31))
* Validate the quantity prop and throw a ConfigurationError if it is larger than the list size ([link](https://github.com/PipedreamHQ/pipedream/pull/6233/files?diff=unified&w=0#diff-ddc1f2837cd6e660224736fb43797210ff92faa88f5290c3fd483952e8cd076bL16-R31))
* Use the splice method to remove and return random items from the list until the quantity is reached ([link](https://github.com/PipedreamHQ/pipedream/pull/6233/files?diff=unified&w=0#diff-ddc1f2837cd6e660224736fb43797210ff92faa88f5290c3fd483952e8cd076bL16-R31))
* Return an array of random items as the result ([link](https://github.com/PipedreamHQ/pipedream/pull/6233/files?diff=unified&w=0#diff-ddc1f2837cd6e660224736fb43797210ff92faa88f5290c3fd483952e8cd076bL16-R31))

### CODE REVIEW

**Comment by pipedream-component-development** [comment]
Thank you so much for submitting this! We've added it to our backlog to review, and our team has been notified.

**Comment by pipedream-component-development** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:
- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

File: components/helper_functions/actions/random-item-from-list/random-item-from-list.mjs [hunk 1]
```diff
@@ -1,10 +1,12 @@
 import helperFunctions from "../../helper_functions.app.mjs";
+import { ConfigurationError } from "@pipedream/platform";
 
 export default {
   key: "helper_functions-random-item-from-list",
-  name: "Random Item from List",
-  description: "Returns a randomly selected value from a user-defined list of options.",
-  version: "0.0.1",
+  name: "Random Item(s) from List",
```
[the remark this answers is not in the document]
Comment (dylburger) [reply components/helper_functions/actions/random-item-from-list/random-item-from-list.mjs:6]: Changed capitalization to sentence case as per Google's style guide

File: components/helper_functions/actions/random-item-from-list/random-item-from-list.mjs [hunk 2]
```diff
@@ -1,10 +1,12 @@
...
+  description:
+    "Returns a randomly selected value(s) from a user-defined list of options.",
```
[the remark this answers is not in the document]
Comment (dylburger) [reply components/helper_functions/actions/random-item-from-list/random-item-from-list.mjs:8]: Changed capitalization to sentence case as per Google's style guide

File: components/helper_functions/actions/random-item-from-list/random-item-from-list.mjs [hunk 3]
```diff
@@ -13,8 +15,19 @@ export default {
       label: "List",
       description: "List of items. Can pass an array from a previous step.",
     },
+    quantity: {
+      type: "integer",
+      label: "Quantity",
```
Comment (dylburger) [inline components/helper_functions/actions/random-item-from-list/random-item-from-list.mjs:20]:
```suggestion
      label: "Quantity",
```

Comment (dylburger) [reply components/helper_functions/actions/random-item-from-list/random-item-from-list.mjs:20]: Changed capitalization to sentence case as per Google's style guide

File: components/helper_functions/actions/random-item-from-list/random-item-from-list.mjs [hunk 4]
```diff
@@ -13,8 +15,19 @@ export default {
...
+      description: "How many items to return.",
+      default: 1,
+      min: 1,
+    },
   },
   run() {
-    return this.list[Math.floor(Math.random() * this.list.length)];
+    if(this.quantity > this.list.length){
+      throw new ConfigurationError("Quantity must be smaller than the list size");
```
Comment (dylburger) [inline components/helper_functions/actions/random-item-from-list/random-item-from-list.mjs:28]:
```suggestion
      throw new ConfigurationError("Quantity must be smaller than the list size");
```

Comment (dylburger) [reply components/helper_functions/actions/random-item-from-list/random-item-from-list.mjs:28]: Changed capitalization to sentence case as per Google's style guide

**Review by alysonturing — APPROVED** [review]
LGTM! Thank you for your valuable contribution!

**Comment by michelle0927** [comment]
Testing passed! Ready for release!

Test report
[Helper_Functions_6233.pdf](https://github.com/PipedreamHQ/pipedream/files/11400553/Helper_Functions_6233.pdf)

**Pull request merged**